File: 1886-minimum-limit-of-balls-in-a-bag/1886-minimum-limit-of-balls-in-a-bag.py
class Solution:
    def minimumSize(self, nums: List[int], maxOperations: int) -> int:

        right = max(nums)
        left = 1

        while left < right:
            mid = (left + right)//2
            total = 0
            for num in nums:
                total += math.ceil(num / mid) - 1

            if total <=maxOperations:
                right = mid
            else:
                left = mid + 1
        
        return left

File: 1886-minimum-limit-of-balls-in-a-bag/1886-minimum-limit-of-balls-in-a-bag.java
class Solution {
    public int minimumSize(int[] nums, int maxOperations) {
        int right = Arrays.stream(nums).max().getAsInt();
        int left = 1;
        
        while(left<right){
            int mid = (left + right)/2;
            int total = 0;
            for (int num : nums) {
                total += (num - 1) / mid;
            }

            if(total<=maxOperations){
                right = mid;
            }
            else{
                left = mid + 1;
            }
        
        }

        return left;
    }
}